crossword-xml-uploader/src/main/scala/com/gu/crossword/Lambda.scala

```diff
@@ -1,27 +1,74 @@
 package com.gu.crossword
 
-import java.util.{ Map => JMap }
-import com.amazonaws.services.lambda.runtime.{ RequestHandler, Context }
-import com.gu.crossword.crosswords.{ CrosswordStore, CrosswordUploader }
+import java.util.{Map => JMap}
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import com.gu.crossword.crosswords._
+import com.gu.crossword.crosswords.models.CrosswordXmlFile
 
-class Lambda
-    extends RequestHandler[JMap[String, Object], Unit]
-    with CrosswordUploader
-    with CrosswordStore {
+import scala.util.{Failure, Success}
+
+trait CrosswordUploaderLambda
+  extends RequestHandler[JMap[String, Object], Unit]
+    with ComposerOps
+    with CrosswordStore
+    with CrosswordConfigRetriever
+    with CrosswordUploader {
+
+  private def doUpload(url: String, streamName: String, crosswordXmlFile: CrosswordXmlFile): Either[(String, Throwable), String] = {
+    val uploadResult = for {
+      rawXml <- uploadCrossword(url)(crosswordXmlFile)
+      crosswordXml <- XmlProcessor.process(rawXml)
+      _ <- createPage(streamName)(crosswordXmlFile.key, crosswordXml)
+    } yield ()
 
-  override def handleRequest(event: JMap[String, Object], context: Context): Unit = {
+    uploadResult match {
+      case Success(_) => Right(crosswordXmlFile.key)
+      case Failure(error) => Left((crosswordXmlFile.key, error))
+    }
+  }
 
-    implicit val config = new Config(context)
+  def handleRequest(event: JMap[String, Object], context: Context): Unit = {
+    val config = getConfig(context)
 
     println("The uploading of crossword xml files has started.")
 
-    for {
-      crosswordXmlFile <- getCrosswordXmlFiles(config)
-    } yield {
-      uploadCrossword(crosswordXmlFile)
+    val crosswordXmlFiles = getCrosswordXmlFiles(config.crosswordsBucketName)
+    println(s"Found ${crosswordXmlFiles.size} crossword file(s) to process")
+
+    val (failures, successes) = crosswordXmlFiles.map { crosswordXmlFile =>
+      doUpload(
+        url = config.crosswordMicroAppUrl,
+        streamName = config.composerCrosswordIntegrationStreamName,
+        crosswordXmlFile = crosswordXmlFile
+      )
+    } partitionMap(identity)
+
+    failures.foreach {
+      case (key, error) =>
+        println(s"Failed to upload crossword ${key} with error: ${error.getMessage}")
+        error.getStackTrace.foreach(println)
```

Comment:
I think prefer `error.printStackTrace()` - it's formatted more as you'd expect from a stack trace and I _hope_ (but haven't tested) that cloudwatch will keep the lines together as a single log entry